### PR TITLE
deepfried_dd docs

### PR DESCRIPTION
**Pull request by Chaldron**
This PR adds the docs for deepfried_dd to the website.

### CODE REVIEW

File: _docs/deepfried_dd.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+Note that Linux will automatically prevent you (usually) from writing to physical devices unless you run `dd` as root (i.e. via sudo).
+
+## Implementing dd
+For this assignment, you will be implementing the `dd` utility in C. Your `dd` implementation will copy data from an input file to an output file in a manner specified by its arguments.
+
+### Background: Blocks
+
+Though "blocks" are a fundamental filesystem concept, this assignment does not rely on heavy filesystems knowledge. For the purposes of this assignment, a *block* is simply a single unit of data. For example, if we run `./dd -i input_file -o output_file -b 128`, we are telling `dd` to copy `input_file` to `output_file`, 128 bytes at a time. Your code should write the first 128 bytes of `input_file` to `output_file`, then the next 128, and so on and so forth in a loop until the `input_file` is exhausted (as opposed to copying the entire file in one go). The arguments of `dd` along with more example usage can be found below.
```
Comment (MaciejKrzysiak) [inline _docs/deepfried_dd.md]: Hmm, I tried reading this pretending I had no background knowledge on blocks. While the example is great and should stay, I don't think this effectively explains blocks.

File: _docs/deepfried_dd.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+You must implement the following arguments from the real `dd`. Since it is $CURRENT_YEAR, we won't be using `dd`'s style of arguments (`if=file`, etc), and instead use the standard style (`-i file`, etc). You **should** use `getopt` to parse these arguments, just as you did in the Shell MP.
+
+* `-i <file>`: input file (defaults to stdin)
+* `-o <file>`: output file (defaults to stdout)
+  * You should create this file if does not already exist.
+* `-b <size>`: block size, the number of bytes copied at a time (defaults to 512)
+* `-c <count>`: total number of blocks copied (defaults to the entire file)
+* `-p <count>`: number of blocks to skip at the start of the input file (defaults to 0)
+* `-k <count>`: number of blocks to skip at the start of the output file (defaults to 0)
```
Comment (MaciejKrzysiak) [inline _docs/deepfried_dd.md]: Maybe we should highlight "blocks" here to emphasize that this isn't in bytes.

Comment (Chaldron) [reply _docs/deepfried_dd.md:42]: Will do.

File: _docs/deepfried_dd.md (hunk 1 above)
Comment (Chaldron) [reply _docs/deepfried_dd.md]: Updated in the the latest commit.

**Pull request merged**